### PR TITLE
Update likes block discoverability

### LINKED ISSUE #33163: Likes: make plugin available in editor, even when the module is not active yet
See Primary issue: #32401

### PR DESCRIPTION
**Pull request by andrii-lysenko**
Fixes https://github.com/Automattic/jetpack/issues/33163

## Proposed changes:

- Always register the Likes editor extension.
- Add a new placeholder when the panel is originally loading.
- Add a placeholder with a button to enable the module when it is disabled.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

* Yes: views of the placeholder (jetpack_editor_likes_enable)
## Testing instructions:

* Spin this branch locally, for example, connect via JT
* Go to Posts > Add New
* Click on the Jetpack plugin sidebar.
* Click on the "Likes and Sharing" panel title.
* Click on the button.
* Verify that the module is enabled and working as expected

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/likes/likes.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+add_action( 'jetpack_register_gutenberg_extensions', __NAMESPACE__ . '\register_plugins' );
+
+/**
+ * Register post types
+ */
+add_action(
+	'rest_api_init',
+	function () {
+		if ( ! \Jetpack::is_module_active( 'likes' ) ) {
+			$post_types = get_post_types( array( 'public' => true ) );
```
Comment (monsieur-z) [inline projects/plugins/jetpack/extensions/blocks/likes/likes.php:39]: I'm curious, why is this needed?

**Review by monsieur-z — APPROVED** [review]
It tests well and code changes look sensible to me.

File: projects/plugins/jetpack/extensions/blocks/likes/likes.php (hunk 1 above)
Comment (andrii-lysenko) [reply projects/plugins/jetpack/extensions/blocks/likes/likes.php:39]: Yep, it's a bit tricky here. I put my thoughts https://github.com/Automattic/jetpack/pull/33226#discussion_r1336433663. Eventually, I think we would need to remove it when find the proper solution.
